marklogic-data-hub-central/ui/src/components/column-selector/column-selector.test.tsx

```diff
@@ -1,24 +1,24 @@
 import React from 'react';
 import ColumnSelector from './column-selector';
-import { render, fireEvent } from '@testing-library/react';
-import {entityPropertyDefinitions, selectedPropertyDefinitions} from "../../assets/mock-data/entity-search";
+import { render, fireEvent, waitForElement } from '@testing-library/react';
+import { entityPropertyDefinitions, selectedPropertyDefinitions } from "../../assets/mock-data/entity-search";
 
-let columns = ['id', 'firstName', 'lastName', 'age', 'phoneNumber.work']
+let columns = ['customerId', 'name', 'nicknames', 'shipping', 'billing']
 
 describe('Column selector component', () => {
 
   test('Verify popover is visible', () => {
-    const { queryByTestId } = render(<ColumnSelector popoverVisibility={true} setPopoverVisibility={jest.fn()} entityPropertyDefinitions={entityPropertyDefinitions} selectedPropertyDefinitions={selectedPropertyDefinitions} setColumnSelectorTouched={jest.fn()} columns={columns}/>);
+    const { queryByTestId } = render(<ColumnSelector popoverVisibility={true} setPopoverVisibility={jest.fn()} entityPropertyDefinitions={entityPropertyDefinitions} selectedPropertyDefinitions={selectedPropertyDefinitions} setColumnSelectorTouched={jest.fn()} columns={columns} primaryKey={''}/>);
```

Comment:
Its better to create a props object that can then be used in `render`. We will then have a common set of props required for the component and can add overrides based on the needs of the test. We can avoid making changes in every test like adding primaryKey in this PR and the tests will be maintainable.

Reply:
Agree it's better to use props objects and override them when needed. Thanks.